server/app/services/ti/TIClientCreationResult.java

```diff
@@ -0,0 +1,42 @@
+package services.ti;
+
+import forms.AddApplicantToTrustedIntermediaryGroupForm;
+import java.util.Optional;
+import play.data.Form;
+
+/**
+ * Holds state relevant to the result of attempting to create a Client for TrustedIntermediary.
+ *
+ * <p>If the creation attempt was successful, the response is a successful object with an empty
+ * form.
+ *
+ * <p>If the creation attempt was not successful, the result contains the same form object with the
+ * form validation errors.
+ */
+public final class TIClientCreationResult {
```

Comment:
issue (blocking): it's still not clear to me why this class is necessary vs returning the `Optional<FormClass>` from the service

Reply:
This is needed in the very next PR i am going to work on. I will be adding a form for Updating DOB and this class will also hold the form object which contains errors

Reply:
In that case the service class will presumably have a new method updateClient which can return an optional of the other form, no?

Reply:
I have deleted this class